extensions/stackdriver/metric/registry.cc

```diff
@@ -34,13 +36,14 @@ MonitoredResource getMonitoredResource(
     const std::string &monitored_resource_type,
     const NodeInfo &local_node_info) {
   google::api::MonitoredResource monitored_resource;
-  monitored_resource.set_type(kContainerMonitoredResource);
+  monitored_resource.set_type(monitored_resource_type);
+  auto platform_metadata = local_node_info.platform_metadata();
```

Comment:
The left side doesn't copy platform_metadata but your PR does the copy. Maybe just because you want to use operator[]?

Reply:
Yes, for code simplicity. This code path is only executed once at initialization so a copy is fine.